mockgcp/mockdiscoveryengine/datastore.go

```diff
@@ -48,6 +48,11 @@ func (s *dataStoreService) CreateDataStore(ctx context.Context, req *pb.CreateDa
 	}
 	now := time.Now()
 
+	if name.Collection != "default_collection" {
+		collectionLink := fmt.Sprintf("projects/%d/locations/global/collections/%s", name.Project.Number, name.Collection)
+		return nil, status.Errorf(codes.NotFound, "Collection with name %s does not exist.", collectionLink)
+	}
+
```

Comment:
Shall we change the `spec.connection` to a reference?

Reply:
This is in mockgcp.

discoveryengine doesn't have gcloud support, so we have to do mockgcp at this point.